src/main/scala/beam/utils/csv/conversion/PlansXml2CsvConverter.scala

```diff
@@ -0,0 +1,99 @@
+package beam.utils.csv.conversion
+
+import java.io.File
+
+import scala.xml.parsing.ConstructingParser
+import scala.xml.{Node, NodeSeq}
+
+object PlansXml2CsvConverter extends Xml2CsvFileConverter {
+
+  override protected def fields: Seq[String] =
+    Seq(
+      "personId",
+      "planId",
+      "planElementType",
+      "activityIndex",
+      "activityType",
+      "locationX",
+      "locationY",
+      "endTime",
+      "mode"
+    )
+
+  private case class PlanFlat(
+    personId: Int,
+    planId: Int,
+    planElementType: String,
+    activityIndex: Int,
+    activityType: String,
+    locationX: Double,
+    locationY: Double,
+    endTime: String,
+    mode: String
+  ) {
+    override def toString: String = {
+      Seq(personId, planId, planElementType, activityIndex, activityType, locationX, locationY, endTime, mode).mkString(
+        FieldSeparator
+      )
+    }
+  }
+
+  private case class ActivityOrLeg(
+    activityIndex: Int,
+    activityType: String,
+    locationX: Double,
+    locationY: Double,
+    endTime: Option[String],
+    mode: String,
+    elementType: String
+  )
+
+  private def toActivity(node: Node, index: Int): ActivityOrLeg = {
+    ActivityOrLeg(
+      activityIndex = index,
+      activityType = Option(node.attributes("type")).map(_.toString.trim).getOrElse(""),
+      locationX = Option(node.attributes("x")).map(_.text.toDouble).getOrElse(0),
+      locationY = Option(node.attributes("x")).map(_.text.toDouble).getOrElse(0),
```

Comment:
Typo here `locationY = Option(node.attributes("x")` should be `locationY = Option(node.attributes("y")`

Reply:
nice catch